repository/process.go

```diff
@@ -1,14 +1,42 @@
 package repository
 
 import (
-	"github.com/google/go-github/v32/github"
+	"time"
+
+	"github.com/google/go-github/v43/github"
 	"github.com/gruntwork-io/git-xargs/config"
+	"github.com/gruntwork-io/git-xargs/types"
 	"github.com/gruntwork-io/go-commons/logging"
 	"github.com/remeh/sizedwaitgroup"
 	"github.com/sirupsen/logrus"
 )
 
-// ProcessRepos loops through every repo we've selected and use a WaitGroup so that the processing can happen in parallel
+// openPullRequestsWithThrottling calls the method to open a pull request after waiting on the internal ticker channel, which
+// reflects the value of the --seconds-between-prs flag
+func openPullRequestsWithThrottling(gitxargsConfig *config.GitXargsConfig, pr types.OpenPrRequest, wg *sizedwaitgroup.SizedWaitGroup) {
+	defer wg.Done()
+
+	logger := logging.GetLogger("git-xargs")
+	logger.Debugf("pullRequestWorker received pull request job. Delay: %d. Retries: %d for repo: %s on branch: %s\n", pr.Delay, pr.Retries, pr.Repo.GetName(), pr.Branch)
+
+	// Space out open PR calls to GitHub API to avoid being aggressively rate-limited
+	// Uses the gitxargsConfig Ticker which is set via default and overridden by the seconds-between-prs flag
+	<-gitxargsConfig.Ticker.C
+	if pr.Delay != 0 {
+		logger.Debugf("Throttled pull request worker delaying %d before attempting to re-open pr against repo: %s", pr.Delay, pr.Repo.GetName())
+		time.Sleep(time.Duration(pr.Delay) * time.Second)
+	}
+	// Make pull request. Errors are handled within the method itself
+	openPullRequest(gitxargsConfig, pr)
+}
+
+// ProcessRepos loops through every repo we've selected and uses a WaitGroup so that the processing can happen in parallel
```

Comment:
Super NIT
```suggestion
// ProcessRepos loops through every repo we've selected and uses a WaitGroup so that the processing can happen in parallel.

Only because the next line is a new sentence.
```